TodoList/Services/ToDo/ToDoService.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using TodoList.DTOs;
+using TodoList.Models;
+
+namespace TodoList.Services.ToDo
+{
+    public class ToDoService : IToDoService
+    {
+        private readonly TodoDbContext _context;
+        private readonly IMapper _mapper;
+
+        public ToDoService(TodoDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+        enum State
+        {
+            NotComplete = 0,
+            Done = 1
+        }
+        public async Task AddNewTask(Guid userId ,ToDoRequest toDo)
+        {
+            var item = _mapper.Map<Todo>(toDo);
+            item.UserId = userId;
+            item.Date = DateTime.Now;
+            item.Status = (int)State.NotComplete;
+            _context.Tasks.Add(item);
+            await _context.SaveChangesAsync();
+        }
+        public async Task CompleteTask(Guid userId, List<Guid> taskIDs)
+        {
+            using (IDbContextTransaction transaction = _context.Database.BeginTransaction())
+            {
+                var task = await _context.Tasks.Where(c => taskIDs.Contains(c.Id)).ToListAsync();
+                foreach (var x in task)
+                {
+                    x.Status = (int)State.Done;
+                }
+                _context.UpdateRange(task);
+                await _context.SaveChangesAsync();
+                await transaction.CommitAsync();
+            }
+        }
+        public async Task DeleteTask(Guid userId, Guid taskId)
+        {
+            var item = _context.Tasks.FirstOrDefault(x => x.UserId == userId && x.Id == taskId);
```

Comment:
check trường hợp item null
```suggestion
            var item = await _context.Tasks.FirstOrDefault(x => x.UserId == userId && x.Id == taskId);
```